PhysicsTools/PatAlgos/python/jetCollectionTools.py

```diff
@@ -0,0 +1,621 @@
+import FWCore.ParameterSet.Config as cms
+
+from Configuration.Eras.Modifier_run2_miniAOD_80XLegacy_cff import run2_miniAOD_80XLegacy
+from Configuration.Eras.Modifier_run2_nanoAOD_94X2016_cff import run2_nanoAOD_94X2016
+
+from RecoJets.JetProducers.PFJetParameters_cfi import PFJetParameters
+from RecoJets.JetProducers.GenJetParameters_cfi import GenJetParameters
+from RecoJets.JetProducers.AnomalousCellParameters_cfi import AnomalousCellParameters
+
+from PhysicsTools.PatAlgos.tools.jetTools import addJetCollection, supportedJetAlgos
+from PhysicsTools.PatAlgos.producersLayer1.jetUpdater_cfi import updatedPatJets
+from PhysicsTools.PatAlgos.recoLayer0.jetCorrFactors_cfi import patJetCorrFactors
+
+from CommonTools.PileupAlgos.Puppi_cff import puppi
+from CommonTools.PileupAlgos.softKiller_cfi import softKiller
+
+from PhysicsTools.NanoAOD.common_cff import Var, P4Vars
+from PhysicsTools.NanoAOD.jets_cff import jetTable
+
+import copy
+import re
+
+
+JETVARS = cms.PSet(P4Vars,
+  HFHEF     = Var("HFHadronEnergyFraction()", float, doc = "energy fraction in forward hadronic calorimeter", precision = 6),
+  HFEMEF    = Var("HFEMEnergyFraction()",     float, doc = "energy fraction in forward EM calorimeter",       precision = 6),
+  area      = jetTable.variables.area,
+  chHEF     = jetTable.variables.chHEF,
+  neHEF     = jetTable.variables.neHEF,
+  chEmEF    = jetTable.variables.chEmEF,
+  neEmEF    = jetTable.variables.neEmEF,
+  muEF      = jetTable.variables.muEF,
+  rawFactor = jetTable.variables.rawFactor,
+  jetId     = jetTable.variables.jetId,
+  jercCHPUF = jetTable.variables.jercCHPUF,
+  jercCHF   = jetTable.variables.jercCHF,
+)
+
+
+#============================================
+#
+# JetInfo
+#
+#============================================
+class GenJetInfo(object):
```

Comment:
BTW, other classes in jetTools have "Note that the tool needs to be derived from ConfigToolBase to be usable in the configEditor."
Should this apply here as well?